pkg/ucp/ucpOrg.go

```diff
@@ -263,10 +265,32 @@ func (c *Client) SetGrant(collection, role, subject string, flags uint) error {
 		return fmt.Errorf("Unknown Grant Type")
 	}
 
-	url := fmt.Sprintf("%s/collectionGrants/%s/%s/%s?type=%s", c.UCPURL, subject, collection, role, grantType)
+	matched, err := regexp.MatchString("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", role)
+	if !matched {
```

Comment:
Perhaps a `log.Debug()` that points out that we're having to delve into a deeper search?

Reply:
Ah good point

Reply:
```
DEBU[0000] Looking up UUID for role: efs-provisioner-runner 
DEBU[0000] [success] HTTP Status code 200               
DEBU[0000] Parsing all roles                            
DEBU[0000] Found UUID: 5d00836b-f9e6-498b-b0df-0a889ab25f3b 
```

Reply:
```
DEBU[0000] Reading Token from [/Users/docker/.ucptoken] 
DEBU[0000] Role appears to already be a UUID: 0a05a837-6d52-4ef0-acf8-7b6e99536ea5 
```